src/testFixtures/java/org/opensearch/knn/TestUtils.java

```diff
@@ -181,6 +186,45 @@ public static double calculateRecallValue(List<List<String>> searchResults, List
         return sum / recalls.size();
     }
 
+    public static PriorityQueue<DistVector> computeGroundTruthValues(int k, SpaceType spaceType, IDVectorProducer idVectorProducer) {
+        PriorityQueue<DistVector> pq = new PriorityQueue<>(k, new DistComparator());
+        int numDocs = idVectorProducer.getVectorCount();
+        float[] queryVector = idVectorProducer.getVector(numDocs);
+
+        for (int id = 0; id < numDocs; id++) {
+            float[] indexVector = idVectorProducer.getVector(id);
+            float dist = computeDistFromSpaceType(spaceType, indexVector, queryVector);
+            pq = sortVectorsUsingDist(pq, k, dist, id);
+        }
+        return pq;
+    }
+
+    public static float computeDistFromSpaceType(SpaceType spaceType, float[] indexVector, float[] queryVector) {
+        float dist = 0.0f;
+        if (spaceType != null) {
+            dist = KNN_SCORING_SPACE_TYPE.getOrDefault(
+                spaceType,
+                (defaultQueryVector, defaultIndexVector) -> throwInvalidInputException(spaceType)
+            ).apply(queryVector, indexVector);
+        }
+        return dist;
+    }
+
+    public static Float throwInvalidInputException(SpaceType spaceType) {
```

Comment:
Is this function used elsewehere outisde of line 207? If not, I dont think it is necessary. 

If it is necessary, why is it returning Float instead of being void?